src/index.ts

```diff
@@ -10,72 +9,79 @@ import { clone } from './git'
 import { getParameter } from './ssm'
 import { parse } from './yaml'
 
+const ecrDockerUrl = new RegExp(/^[^\.]+\.dkr\.ecr\.[^\.]+\.amazonaws\.com/)
 const isPost = getState('isPost') === 'true'
 
 if (!isPost) {
-  target().then(target => {
-    // Run Container Directly
-    if (!target.clone) return target.dockerImage().then(runDocker)
+	target()
+		.then(async target => {
+			// Run Container Directly
+			if (!target.clone) {
+				return target.dockerImage().then(runDocker)
+			}
 
-    return token().then(GITHUB_TOKEN => {
-      const repo = `https://${GITHUB_TOKEN}@github.com/${(target.url as NodeURL).action}.git`
+			const dir = await token()
+				.then(githubToken => `https://${githubToken}@github.com/${(target.url as NodeURL).action}.git`)
+				.then(repo => group('Cloning Target Action', () => clone(repo)))
 
-      return group('Cloning Target Action', () => clone(repo))
-        .then(dir => {
-          const readFile = promisify(readFileCallback)
+			return fs.promises
+				.readFile(join(dir, 'action.yml'))
+				.then(text => parse(text.toString()))
+				.then(async action => {
+					if (action.isNode()) {
+						group('Running Node Action', () => {
```

Comment:
You dropped the return  here... without it the Promise from exec is not bubbled up